scipy/stats/_stats_py.py

```diff
@@ -2849,8 +2858,11 @@ def obrientransform(*samples):
     if sLast:
         for arr in arrays[:-1]:
             if sLast != arr.shape:
-                return np.array(arrays, dtype=object)
-    return np.array(arrays)
+                if is_numpy(xp):
+                    return xp.array(arrays, dtype=object)
+                else:
+                    return arrays
+    return xp.stack(arrays)
```

Comment:
I know the plan is to immediately change this, but I I still fill that in the meantime the switch should be `SCIPY_ARRAY_API`. At least we should not return a single `xp` array when possible and a tuple of `xp`-arrays otherwise because an `xp`-array can't necessarily even be unpacked.